tests/sentence_transformers/test_training_nli.py

```diff
@@ -0,0 +1,121 @@
+    eval_dataset = load_dataset("sentence-transformers/all-nli", "pair-class", split="dev").select(range(1000))
+    logging.info(train_dataset)
+    
+    # 3. Define our training loss: https://sbert.net/docs/package_reference/sentence_transformer/losses.html#softmaxloss
+    train_loss = losses.SoftmaxLoss(
+        model=model,
+        sentence_embedding_dimension=model.get_sentence_embedding_dimension(),
+        num_labels=3,
+    )
+    
+    # 4. Define an evaluator for use during training. This is useful to keep track of alongside the evaluation loss.
+    stsb_eval_dataset = load_dataset("sentence-transformers/stsb", split="validation")
+    dev_evaluator = EmbeddingSimilarityEvaluator(
+        sentences1=stsb_eval_dataset["sentence1"],
+        sentences2=stsb_eval_dataset["sentence2"],
+        scores=stsb_eval_dataset["score"],
+        main_similarity=SimilarityFunction.COSINE,
+        name="sts-dev",
+    )
+    logging.info("Evaluation before training:")
+    #dev_evaluator(model)
+    
+    # 5. Define the training arguments
+    args = SentenceTransformerGaudiTrainingArguments(
+        # Required parameter:
+        output_dir=output_dir,
+        # Optional training parameters:
+        num_train_epochs=1,
+        per_device_train_batch_size=train_batch_size,
+        per_device_eval_batch_size=train_batch_size,
+        warmup_ratio=0.1,
+        #fp16=True,  # Set to False if you get an error that your GPU can't run on FP16
+        #bf16=False,  # Set to True if you have a GPU that supports BF16
+        # Optional tracking/debugging parameters:
+        #evaluation_strategy="steps",
+        eval_steps=100,
+        save_strategy="steps",
+        save_steps=100,
+        save_total_limit=2,
+        logging_steps=100,
+        run_name="nli-v1",  # Will be used in W&B if `wandb` is installed
+        use_habana = True,
+        gaudi_config_name = 'Habana/bert-base-uncased',
+        use_lazy_mode = True,
+        use_hpu_graphs = True,
+        use_hpu_graphs_for_inference = False,
+        use_hpu_graphs_for_training = True,
+    )
+    
+    # 6. Create the trainer & start training
+    trainer = SentenceTransformerGaudiTrainer(
+        model=model,
+        args=args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+        loss=train_loss,
+        evaluator=dev_evaluator,
+    )
+    trainer.train()
+    
+    # 7. Evaluate the model performance on the STS Benchmark test dataset
+    test_dataset = load_dataset("sentence-transformers/stsb", split="test")
+    test_evaluator = EmbeddingSimilarityEvaluator(
+        sentences1=test_dataset["sentence1"],
+        sentences2=test_dataset["sentence2"],
+        scores=test_dataset["score"],
+        main_similarity=SimilarityFunction.COSINE,
+        name="sts-test",
+    )
+    test_evaluator(model)
+    
+    # 8. Save the trained & evaluated model locally
+    #final_output_dir = f"{output_dir}/final"
+    #model.save(final_output_dir)
```

Comment:
Thanks @regisss for the merge! Also thanks for the help from @hchauhan123  @mandy-li  @pi314ever on this PR! 

About the comment above it could be uncomment. I thought normally CI/CD will not have big harddisk capacity to save lots of output then i commented it. This save command will save the different folder version each time which may gradually have some problem on the test disk. Just for this concern i commented. 

Thanks!

Reply:
@ZhengHongming888 , other tests use something like `with tempfile.TemporaryDirectory() as tmpdirname` to circumvent the disk issue.

Reply:
Thanks @vidyasiv  i will correct them in next PR together. thanks.